diff --git a/src/Plugin/search_api/backend/SearchApiElasticsearchBackend.php b/src/Plugin/search_api/backend/SearchApiElasticsearchBackend.php
index 90c1ec0..0d1b617 100644
--- a/src/Plugin/search_api/backend/SearchApiElasticsearchBackend.php
+++ b/src/Plugin/search_api/backend/SearchApiElasticsearchBackend.php
@@ -732,8 +732,8 @@ class SearchApiElasticsearchBackend extends BackendPluginBase implements PluginF
     foreach ($facets as $key => $facet) {
       $terms = [];
 
-      // Handle 'and' operator.
-      if ($facet['operator'] == 'and' || ($facet['operator'] == 'or' && !isset($response['aggregations'][$key . '_global']))) {
+      // Handle '=' operator.
+      if ($facet['operator'] == '=' || ($facet['operator'] == 'or' && !isset($response['aggregations'][$key . '_global']))) {
         if (!empty($facet['type']) && $facet['type'] == 'stats') {
           $terms = $response['aggregations'][$key];
         }
