diff --git a/src/Plugin/views/field/CurrentStateViewsField.php b/src/Plugin/views/field/CurrentStateViewsField.php
index e0e376e..9230951 100644
--- a/src/Plugin/views/field/CurrentStateViewsField.php
+++ b/src/Plugin/views/field/CurrentStateViewsField.php
@@ -100,11 +100,12 @@ class CurrentStateViewsField extends FieldPluginBase implements TrustedCallbackI
    */
   public function render(ResultRow $values) {
     $entity = $values->_entity;
+    $entity_langcode = $values->node_field_data_langcode;
     $storage = $this->entityTypeManager->getStorage($entity->getEntityTypeId());
-    $get_latest_revision_id = $storage->getLatestRevisionId($entity->id());
+    $get_latest_revision_id = $storage->getLatestTranslationAffectedRevisionId($entity->id(), $entity_langcode);
     $get_latest_revision_entity = $storage->loadRevision($get_latest_revision_id);
     if (!empty($get_latest_revision_entity)) {
-      $entity = $get_latest_revision_entity;
+      $entity = $get_latest_revision_entity->hasTranslation($entity_langcode) ? $get_latest_revision_entity->getTranslation($entity_langcode) : $get_latest_revision_entity;
     }
     if (!empty($entity->moderation_state->value)) {
       $current_state = $this->moderationInformation->getWorkflowForEntity($entity)->getTypePlugin()->getState($entity->moderation_state->value)->label();
